### PR TITLE
Issue/194 Use default value for nil adjust_mw 

### LINKED ISSUE #194: Adjust mw nan
## Detailed Description
If the adjust mw is Nan filled in with zero

This can be done here - https://github.com/openclimatefix/nowcasting_datamodel/blob/main/nowcasting_datamodel/models/forecast.py#L331

## Context
- some times there are nan if we dont know what to adjust, this protects against that

## Possible Implementation
see link above, `if np.nan then adjust_mw=0`

### PR DESCRIPTION
**Pull request by petermnhull**
# Pull Request

## Description

Fixes #194 

Also includes a few minor additional improvements:
- [Fix flake8 errors in tests, mainly due to unused imports](https://github.com/openclimatefix/nowcasting_datamodel/commit/9db86bc3cdd28aa3a1adba6810023b647073541d)
- [Fix typing errors which were getting caught in linting locally](https://github.com/openclimatefix/nowcasting_datamodel/commit/27a38c24be3aa2cc9c614aede9cae17b28819bf2)
- Fix some typos in comments.
- [Fix BST-dependent tests in test_read.](https://github.com/openclimatefix/nowcasting_datamodel/commit/fc163074584330193c48c4a4ff93eb743f077d40) This one I'm a bit less sure on, but I believe there is a timezone dependant test here which relies on non-BST timing which failed locally until I removed the timezones. See errors below, let me know if there's something I'm missing here.

```
FAILED tests/read/test_read.py::test_get_latest_input_data_last_updated - assert datetime.datetime(2023, 7, 1, 18, 29, 57, 40344) == datetime.datetime(2023, 7, 1, 17, 29, 57, 40344)
FAILED tests/read/test_read.py::test_update_latest_input_data_last_updated - assert datetime.datetime(2023, 7, 1, 18, 29, 57, 387618) == datetime.datetime(2023, 7, 1, 17, 29, 57, 38...
```

## How Has This Been Tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.
Please also list any relevant details for your test configuration_

Tested this locally on bare metal in a virtualenv.

- [x] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [x] Yes / NA

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: nowcasting_datamodel/models/forecast.py [hunk 1]
```diff
@@ -327,10 +327,14 @@ def from_orm(cls, obj: ForecastValueSQL):
 
         # this is because from orm doesnt copy over '_' variables.
         # But we don't want to expose this in the API
+        default_value = 0.0
         if hasattr(obj, "adjust_mw"):
-            m._adjust_mw = obj.adjust_mw
+            adjust_mw = obj.adjust_mw
+            if not adjust_mw:
```
Comment (peterdudfield) [inline nowcasting_datamodel/models/forecast.py:333]: should this be` if np.isnan(adjust_mw)`

File: tests/models/test_models.py [hunk 2]
```diff
@@ -122,6 +122,40 @@ def test_forecast_latest_to_pydantic(forecast_sql):
...
+    f._adjust_mw = np.nan
+
+    actual = ForecastValue.from_orm(f)
+    expected = ForecastValue(
+        target_time=datetime(2023, 1, 1, 0, 30, tzinfo=timezone.utc),
+        expected_power_generation_megawatts=1.0,
+        expected_power_generation_normalized=None,
+    )
+    assert actual == expected
+    assert actual._adjust_mw == 0.0
```
Comment (peterdudfield) [inline tests/models/test_models.py:156]: nice thanks for this

File: nowcasting_datamodel/models/forecast.py (hunk 1 above)
Comment (peterdudfield) [reply nowcasting_datamodel/models/forecast.py:333]: Maybe its the same thing, but what i suggested it perhaps more readable

**Review by peterdudfield — APPROVED** [review]
Thanks for doing all this, and fixing lots of linting and spelling

ive left one small comment

File: nowcasting_datamodel/models/forecast.py (hunk 1 above)
Comment (petermnhull) [reply nowcasting_datamodel/models/forecast.py]: Good call, thanks.

Unfortunately `np.isnan` doesn't cover the case where `adjust_mw` gets set to `None` by Pydantic when it's completely missing, giving a `TypeError`. But I'm not sure if a `None` _can_ ever sneak in here in prod. Added an `np.isnan` check to cover all scenarios as well as the `None` check, but let me know if you'd prefer just leaving out the latter.

Comment (peterdudfield) [reply nowcasting_datamodel/models/forecast.py]: Cool- that looks good - thank you. I prefer leaving it in - thanks

**Pull request merged**